dffml/base.py

```diff
@@ -235,7 +259,61 @@ def config(cls):
     """
     Decorator to create a dataclass
     """
-    datacls = dataclasses.dataclass(eq=True, init=True)(cls)
+    class_list = []
+    newdict = {}
+    defaultdict = {}
+    if len(inspect.getmro(cls)) > 2:
+        class_list = list(inspect.getmro(cls)[:-1])
+    else:
+        class_list = [cls]
+    for clss in class_list:
+        cls_annotations = clss.__dict__.get("__annotations__", {})
+        cls_dict = clss.__dict__
+        for var in cls_annotations:
+            if not dataclasses.is_dataclass(clss):
+                if cls_dict.get(var) and isinstance(
+                    cls_dict.get(var), dataclasses.Field
+                ):
+                    if "dataclasses._MISSING_TYPE" in repr(
+                        cls_dict.get(var).default
+                    ) and "dataclasses._MISSING_TYPE" in repr(
+                        cls_dict.get(var).default_factory
+                    ):
+                        newdict[var] = cls_annotations[var]
+            else:
+                if clss.__dataclass_fields__.get(var):
+                    if "dataclasses._MISSING_TYPE" in repr(
+                        clss.__dataclass_fields__.get(var).default
+                    ) and "dataclasses._MISSING_TYPE" in repr(
+                        clss.__dataclass_fields__.get(var).default_factory
+                    ):
+                        newdict[var] = cls_annotations[var]
+
+        for var in cls_annotations:
+            if not dataclasses.is_dataclass(clss):
+                if cls_dict.get(var) and isinstance(
+                    cls_dict.get(var), dataclasses.Field
+                ):
+                    if "dataclasses._MISSING_TYPE" not in repr(
+                        cls_dict.get(var).default
+                    ) or "dataclasses._MISSING_TYPE" not in repr(
+                        cls_dict.get(var).default_factory
+                    ):
+                        defaultdict[var] = cls_annotations[var]
+            else:
+                if clss.__dataclass_fields__.get(var):
+                    if "dataclasses._MISSING_TYPE" not in repr(
+                        clss.__dataclass_fields__.get(var).default
+                    ) or "dataclasses._MISSING_TYPE" not in repr(
+                        clss.__dataclass_fields__.get(var).default_factory
+                    ):
+                        defaultdict[var] = cls_annotations[var]
+
+    defaultdict.update(cls.__annotations__)
+    newdict.update(defaultdict)
+    cls.__annotations__ = newdict
+
+    datacls = dataclasses.dataclass(eq=True, init=False)(cls)
```

Comment:
I feel like you may have explained, but why was it that `init` must equal `False`?

Reply:
What happens is that the `__init__` method is not called before [this line](https://github.com/intel/dffml/blob/master/dffml/base.py#L238), so the variables from the base class are not yet in the `__dict__`, so I took the fields from base dataclasses and added them to `__annotations__` of the derived dataclass and changed `init=False` so that same non default argument error would not be thrown and tried other things like adding directly to the `__dict__` of derived class which didn't work either. I guess `init=True` overrides all that extra code.